profiling/src/profile/mod.rs

```diff
@@ -553,13 +582,19 @@ impl Profile {
         let mut local_root_span_id_label_offset: Option<usize> = None;
         for label in sample.labels.iter() {
             let key = self.intern(label.key);
-            let str = label.str.map(|s| self.intern(s)).unwrap_or(0);
-            let num_unit = label.num_unit.map(|s| self.intern(s)).unwrap_or(0);
+            let str = label
+                .str
+                .map(|s| self.intern(s))
+                .unwrap_or(StringId::zero());
+            let num_unit = label
+                .num_unit
+                .map(|s| self.intern(s))
+                .unwrap_or(StringId::zero());
 
             if key == self.endpoints.local_root_span_id_label {
                 // Panic: if the label.str isn't 0, then str must have been provided.
                 anyhow::ensure!(
-                    str == 0,
+                    str.is_zero(),
```

Comment:
Ah, I was a little confused why you added a special `is_zero` method, but this makes sense.  0 is special as far as these IDs go.